### ISSUE TITLE
keepAlive() assumes System.nanoTime() is monotonically increasing, causing assertion failure

### ISSUE THREAD

**Issue by craffit** [akka/akka#31061]
On our environment we triggered the following issue:

```
{"log":"java.lang.AssertionError: assertion failed: Timer should have triggered only after deadline but now is 2156629372696 and deadline was 2156633393892 diff -4021196.\n","stream":"stdout","time":"2022-01-13T00:44:15.711461312Z"}
2022-01-13 01:44:15	
{"log":"\u0009at scala.Predef$.assert(Predef.scala:223)\n","stream":"stdout","time":"2022-01-13T00:44:15.711466297Z"}
2022-01-13 01:44:15	
{"log":"\u0009at akka.stream.impl.Timers$IdleInject$$anon$7.onTimer(Timers.scala:280)\n","stream":"stdout","time":"2022-01-13T00:44:15.711470557Z"}
2022-01-13 01:44:15	
...
```

Looking at https://stackoverflow.com/a/8854104 it seems that the assumption that is being made here that nanoTime() is monotonically increasing does not always hold. https://github.com/akka/akka/commit/01603ccc90befb5f7ce27f3c7e240f54f7ce675f brought in that assumption.

**Comment by johanandren** [comment]
I think this is a duplicate of #30333

**Comment by johanandren** [comment]
For the record the bug about nanoTime not being montonic is marked as affecting Java 5 and 6 which is not supported by current Akka versions: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6458294

**Comment by craffit** [comment]
This was triggered on java 11.

Ah sorry for filing the duplicate

**Comment by craffit** [comment]
duplicate
